wallet/notifications.go

```diff
@@ -198,6 +198,57 @@ func makeTxSummary(dbtx walletdb.ReadTx, w *Wallet, details *udb.TxDetails) Tran
 	}
 }
 
+func makeTicketSummary(dbtx walletdb.ReadTx, w *Wallet, details *udb.TicketDetails) *TicketSummary {
+	var ticketStatus = TicketStatusLive
+
+	ticketTransactionDetails := makeTxSummary(dbtx, w, details.Ticket)
+	if details.Spender != nil {
+		spenderTransactionDetails := makeTxSummary(dbtx, w, details.Spender)
+		if details.Spender.TxType == stake.TxTypeSSGen {
+			ticketStatus = TicketStatusVoted
+		} else if details.Spender.TxType == stake.TxTypeSSRtx {
+			ticketStatus = TicketStatusRevoked
+		}
+		return &TicketSummary{
+			Ticket:  &ticketTransactionDetails,
+			Spender: &spenderTransactionDetails,
+			Status:  ticketStatus,
+		}
+	}
+
+	if details.Ticket.Height() == int32(-1) {
+		ticketStatus = TicketStatusUnmined
+	} else {
+		txmgrNs := dbtx.ReadBucket(wtxmgrNamespaceKey)
+
+		_, tipHeight := w.TxStore.MainChainTip(txmgrNs)
+		ticketAge := int64(tipHeight - details.Ticket.Height())
```

Comment:
rather than calculating the age this way to determine if the ticket is immature or expired, do a check on the number of confirmations.  This is the way other code in the wallet does this check.  There's also an off by one to be aware of.  Grep the wallet package for other uses of TicketMaturity and TicketExpiry to find the correct check.